backend/kernelCI_app/management/commands/process_pending_aggregations.py

```diff
@@ -0,0 +1,425 @@
+    existing_processed: set[ProcessedHardwareStatus],
+    new_processed_entries: set[ProcessedHardwareStatus],
+) -> bool:
+    """Process test status and update status record if not already processed."""
+    to_process = ProcessedHardwareStatus(
+        hardware_key=h_key,
+        entity_id=test.test_id,
+        entity_type=HardwareStatusEntityType.TEST,
+    )
+
+    if to_process in existing_processed:
+        return False
+
+    t_pass, t_fail, t_inc = map_simplified_status_to_count(test.status)
+
+    if test.is_boot:
+        status_record["boot_pass"] += t_pass
+        status_record["boot_failed"] += t_fail
+        status_record["boot_inc"] += t_inc
+    else:
+        status_record["test_pass"] += t_pass
+        status_record["test_failed"] += t_fail
+        status_record["test_inc"] += t_inc
+
+    new_processed_entries.add(to_process)
+
+    return True
+
+
+def _process_build_status(
+    build: Builds,
+    h_key: bytes,
+    status_record: HardwareStatusRecord,
+    existing_processed: set[ProcessedHardwareStatus],
+    new_processed_entries: set[ProcessedHardwareStatus],
+) -> None:
+    """Process build status and update status record if not already processed."""
+
+    """
+    We can't filter out dummy builds before this step,
+    because although we don't count them in build_* status,
+    there are tests that are associated with them, and those tests are counted in test_* status.
+    """
+    if build.id.startswith(MAESTRO_DUMMY_BUILD_PREFIX):
+        return
+
+    to_process = ProcessedHardwareStatus(
+        hardware_key=h_key,
+        entity_id=build.id,
+        entity_type=HardwareStatusEntityType.BUILD,
+    )
+
+    if to_process in existing_processed:
+        return
+
+    if to_process in new_processed_entries:
+        return
+
+    b_pass, b_fail, b_inc = map_simplified_status_to_count(
+        simplify_status(build.status)
+    )
+    status_record["build_pass"] += b_pass
+    status_record["build_failed"] += b_fail
+    status_record["build_inc"] += b_inc
+
+    new_processed_entries.add(to_process)
+
+
+def aggregate_hardware_status(
+    tests_instances: Sequence[PendingTest],
+    builds_by_id: dict[str, Builds],
+) -> tuple[
+    dict[tuple[str, str, str], HardwareStatusRecord], set[ProcessedHardwareStatus]
+]:
+    """
+    Aggregate hardware status from pending tests, builds, and checkouts.
+    Returns a dictionary of hardware status records
+    keyed by (test_origin, platform, checkout_id)
+    and a set of new processed entries.
+    """
+    hardware_status_data: dict[tuple[str, str, str], HardwareStatusRecord] = {}
+    new_processed_entries: set[ProcessedHardwareStatus] = set()
+
+    contexts, keys_to_check = _collect_test_contexts(tests_instances, builds_by_id)
+
+    if not contexts:
+        return hardware_status_data, new_processed_entries
+
+    existing_processed = _get_existing_processed(keys_to_check)
+
+    for test, build, checkout, h_key in contexts:
```

Comment:
here you will get `test_h_key` and `build_h_key`, then be careful when calling `_process_*_status()`